scispacy/abbreviation.py

```diff
@@ -61,17 +61,17 @@ def find_abbreviation(
         long_index -= 1
         short_index -= 1
 
-    # If we complete the string, we end up with -1 here,
-    # but really we want all of the text.
-    long_index = max(long_index, 0)
+    # The last subtraction takes us on to a space character
```

Comment:
This change works because we only ever reach this code if we do complete the string - it's also not quite the case that it takes us onto a space, because here we are operating over words which don't have spaces. I think the original comment is a little clearer?

Reply:
we're not operating over text without whitespace here right? because the long/short form are defined above as space joined text tokens.

Reply:
which makes me notice that the indexing is actually over these single space joined tokens, and that will need to be unified with the below code, rather than just using `text_with_ws`

Reply:
I think to match this, below I will add `len(word.text) + 1` for each word

Reply:
oh actually when there are multiple spaces, spacy give the space its own token.

Reply:
Right, but if we "go off the front" of the string, it won't be a whitespace char (it will refer to nothing as we are just indexing a string and not the doc), which made your comment a little confusing for me. It's a small thing.

Reply:
Oh i see, let me clarify that there are two cases there.

Reply:
Yeah, I think this whole thing would be easier if we moved it to operate on tokens rather than strings, it would reduce how annoying this is in the edge cases, you're right